docs: add TapSDK Versions and Maintenance Cycle for io (#639)
taptap/tds-docs: i18n/en/docusaurus-plugin-content-docs/current/sdk/start/ver-lifetime.mdx

---
title: TapSDK Versions and Maintenance Cycle
sidebar_position: 9
---

import CodeBlock from "@theme/CodeBlock";
import sdkVersions from "/src/docComponents/sdkVersions";
import { Conditional } from "/src/docComponents/conditional";

## Rules for versioning

> TapSDK is the generic name for many of the services we provide, to make it easier for developers to access them on demand, we use a separate module for each sub-service to provide services to the outside world, for example: `TapTap Login` corresponds to the `TapLogin` module, `Announcement System` corresponds to the `TapBillboard` module, and so on.

Reviewer: ```suggestion
TapSDK is the generic name for many of the services we provide. To make it easier for developers to access them on demand, we use a separate module for each sub-service. For example: `TapTap Login` corresponds to the `TapLogin` module, `Billboard` corresponds to the `TapBillboard` module, and so on.
```

> We use [semantic version number](https://semver.org/) to plan the version for TapSDK, the version format is divided into three main parts: `Primary version number. Secondary version number. Revision number', where:

Reviewer: ```suggestion
We use [semantic versioning](https://semver.org/) to plan versions for TapSDK. The version is formatted mainly with three parts: `MAJOR.MINOR.PATCH`, where:
```

> We use [semantic version number](https://semver.org/lang/zh-CN/) to plan the version for TapSDK, the version format is divided into three main parts: `Primary version number. Secondary version number. Revision number', where:

Reviewer: 这里要改成对应文档的英文版。

> - Master Version Number: Indicates a collection of characteristics and API interfaces, if there are incompatible changes introduced in an update, we will upgrade the Master Version Number;

Reviewer: ```suggestion
- MAJOR: Indicates a collection of features and API interfaces. If there are incompatible changes introduced in an update, we will upgrade the MAJOR version number;
```

> - Sub-version numbering: To ensure backward compatibility for functional additions, the sub-version number will be increased. For instance, if a new customer service module is added to version 3.4.0, the updated version will be released as 3.5.0;

Reviewer: ```suggestion
- MINOR: When there are functional additions that ensure backward compatibility, the MINOR version number will be increased. For instance, when a new customer service module is added to version 3.4.0, the updated version will be 3.5.0;
```

> - Version number: If we fix a bug in the SDK or optimize its internal implementation while keeping the external interface unchanged, we will upgrade the version number;

Reviewer: ```suggestion
- PATCH: If we fix a bug in the SDK or optimize its internal implementation while keeping the external interface unchanged, we will upgrade the PATCH version number;
```

> - For advanced releases, and due to platform-specific limitations, we sometimes add the platform information to the "Major version. Revision number" extension;

Reviewer: ```suggestion
- When there are pre-released versions or versions released due to platform-specific limitations, we sometimes append the information to "MAJOR.MINOR.PATCH" as an extension.
```

> Currently the main line of our development is the **3.x version**, for the list of SDK versions and changes for each platform please refer to the following link:

Reviewer: ```suggestion
Currently the main line of our development is the **3.x version**. For the SDK versions and changelist for each platform, please refer to the following links:
```

> - Unity Engine SDK：https://github.com/taptap/TapSDK-Unity/releases

Reviewer: ```suggestion
- Unity Engine SDK: https://github.com/taptap/TapSDK-Unity/releases
```

需要使用英文标点。下同。

- Unreal Engine SDK：https://github.com/taptap/TapSDK-UE4/releases
- Android 原生 SDK：https://github.com/taptap/TapSDK-Android/releases

> - iOS 原生 SDK：https://github.com/taptap/TapSDK-iOS/releases

Reviewer: 这里没翻译。

## Lifecycle of version maintenance

> Each major release is **continuously maintained** if it is a mainline release that we develop/maintain - this is the case for the current 3.x version of the SDK**. If it's not a major release, we **maintain it for two years** after we stop developing new features. If the latest version is 3.16.5, and in the next iteration we change the external interfaces and move to 4.x, then we will maintain 3.x for two years from the time 4.0.0 is released - bug fixes, no new features; and then after two years 3.x will be completely deprecated.

Reviewer: ```suggestion
Each major release is **continuously maintained** if it is a mainline release that we develop/maintain - this is the case for the current 3.x version of the SDK**. If it's not a major release, we **maintain it for two years** after we stop developing new features. If the latest version is 3.16.5, and in the next iteration we change the external interfaces and move to 4.x, then we will maintain 3.x for two years from the time 4.0.0 is released - there will only be bug fixes and no new features; and then after two years 3.x will be completely deprecated.
```

> For minor versions below the current mainline (e.g. 3.1.x/3.2.x), if a bug is found, we will fix it in the current latest version (not in the minor version where the bug was found). Since the interfaces are fully compatible between minor version numbers and the sub-functions are mostly released as independent modules (no impact between modules), developers** can upgrade to the latest minor version number without any burden. developers** can upgrade to the latest minor version** without any burden.

Reviewer: ```suggestion
For minor versions under the current mainline (e.g. 3.1.x/3.2.x), if a bug is found, we will fix it in the current latest version (not in the minor version where the bug was found). Since the interfaces are fully compatible between minor version numbers and the sub-functions are mostly released as independent modules (no impact between modules), developers **can upgrade to the latest minor version without any burden**.

> We can look at a practical example. For example, a developer uses real name authentication and anti-addiction functionality, mainly by accessing the `TapLogin/BootStrap/AntiAddiction` modules, which were accessed with version `3.5.3`, while the latest version is currently iterated to `3.10.4`. Later:

Reviewer: ```suggestion
Let's look at an example. A developer uses real name authentication and anti-addiction by integrating the `TapLogin/BootStrap/AntiAddiction` modules from the version `3.5.3`, while the latest version is currently `3.10.4`. Later:
```

> - Developers found and reported a new bug in the SDK, we followed up to fix it and release version `3.10.5`, developers can directly upgrade the corresponding module version from `3.5.3` to `3.10.5`;

Reviewer: ```suggestion
- The developer found and reported a new bug in the SDK. We followed up to fix it and released version `3.10.5`. The developer can directly upgrade the corresponding modules from `3.5.3` to `3.10.5`;
```

> - If a developer finds and reports a bug that has already been fixed (e.g. no longer exists since `3.8.0`), then the developer can simply upgrade to our latest version, `3.10.4`;

Reviewer: ```suggestion
- If the developer found and reported a bug that has already been fixed (e.g. no longer exists since `3.8.0`), then the developer can simply upgrade to our latest version, `3.10.4`.
```
